### PR TITLE
Add documentation to MQTT design page

### PR DESCRIPTION
**Pull request by gordonwang0**

### CODE REVIEW

File: doc/lib/mqtt.txt [hunk 1]
```diff
@@ -207,9 +207,42 @@ This application is expected to have a peak runtime memory usage of about 600 by
 @page mqtt_design Design
 @brief Architecture behind the MQTT library.
 
-MQTT relies on the [task pool library](@ref taskpool) to process MQTT operations, such as SUBSCRIBE, PUBLISH, or keep-alive. The MQTT API [functions](@ref mqtt_functions) post a job to the task pool, which is transmitted from a background task. Incoming responses are received through a [network receive callback](@ref platform_network_function_receivecallback), which posts another job to process the response to the task pool.
+@section mqtt_design_operations Asynchronous and synchronous operations
+@brief This library provides both asynchronous and synchronous MQTT operations. Asynchronous operations functions end with <b>Async</b>, and their synchronous variants end with <b>Sync</b>.
+
+For example, @ref mqtt_function_publishasync and @ref mqtt_function_publishsync are the respective asynchronous and synchronous MQTT PUBLISH functions. The CONNECT and DISCONNECT operation only have synchronous functions.
```
Comment (muneebahmed10) [inline doc/lib/mqtt.txt]: > CONNECT and DISCONNECT operation

Should be plural

File: doc/lib/mqtt.txt [hunk 2]
```diff
@@ -207,9 +207,42 @@ This application is expected to have a peak runtime memory usage of about 600 by
 @page mqtt_design Design
 @brief Architecture behind the MQTT library.
 
-MQTT relies on the [task pool library](@ref taskpool) to process MQTT operations, such as SUBSCRIBE, PUBLISH, or keep-alive. The MQTT API [functions](@ref mqtt_functions) post a job to the task pool, which is transmitted from a background task. Incoming responses are received through a [network receive callback](@ref platform_network_function_receivecallback), which posts another job to process the response to the task pool.
+@section mqtt_design_operations Asynchronous and synchronous operations
+@brief This library provides both asynchronous and synchronous MQTT operations. Asynchronous operations functions end with <b>Async</b>, and their synchronous variants end with <b>Sync</b>.
```
Comment (muneebahmed10) [inline doc/lib/mqtt.txt]: > Asynchronous operations functions

operations -> operation

File: doc/lib/mqtt.txt [hunk 3]
```diff
@@ -207,9 +207,42 @@ This application is expected to have a peak runtime memory usage of about 600 by
...
+
+@section mqtt_design_asyncoperation Operation diagram
+@brief The following diagram shows the workflow of an operation.
+
+MQTT relies on the [task pool library](@ref taskpool) to process asynchronous MQTT operations in the background. The MQTT API [functions](@ref mqtt_functions) post a job to the task pool, which is transmitted from a background task. Incoming responses are received through a [network receive callback](@ref platform_network_function_receivecallback), which posts another job to process the response to the task pool.
+
+Synchronous operations are implemented as a call to an asynchronous operation. The synchronous function then waits on a semaphore until it receives a notification of completion. The workflow of a synchronous operation is otherwise identical to an asynchronous operation.
```
Comment (muneebahmed10) [inline doc/lib/mqtt.txt]: Nit: Maybe change `as a call to an` to `as calls to a corresponding`

File: doc/lib/mqtt.txt (hunk 1 above)
Comment (gordonwang0) [reply doc/lib/mqtt.txt]: changed

File: doc/lib/mqtt.txt (hunk 2 above)
Comment (gordonwang0) [reply doc/lib/mqtt.txt]: changed

File: doc/lib/mqtt.txt (hunk 3 above)
Comment (gordonwang0) [reply doc/lib/mqtt.txt]: changed

**Pull request merged**